src/rust/src/contract.rs

```diff
@@ -545,3 +545,14 @@ pub(crate) fn compute_write_fee_per_1kb(
 ) -> i64 {
     host_compute_write_fee_per_1kb(bucket_list_size, &fee_config.into())
 }
+
+pub(crate) fn can_parse_transaction(xdr: &CxxBuf) -> bool {
+    let res = TransactionEnvelope::read_xdr(&mut xdr::Limited::new(
+        Cursor::new(xdr.data.as_slice()),
+        Limits {
+            depth: MARSHALLING_STACK_LIMIT,
```

Comment:
The xdrpp value and `MARSHALLING_STACK_LIMIT` must always be equivalent now. It would be safer if `MARSHALLING_STACK_LIMIT` was set from xdrpp instead of being hardcoded.

Reply:
Done.

Reply:
It looks like `MARSHALLING_STACK_LIMIT` is still used for the other limits. I'm guessing this was done because it'd be quite a bit of work to change this and we wouldn't gain much, but should we open an issue for this? It's kinda confusing as is.

Reply:
I don't think that's an issue in the other cases. Here it's kind of necessary to ensure the decoding is consistent between env and core, but in the other cases it's only about applying the transaction on the env side only, so the worse that can happen if we change only the core value is that some edge case transaction would fail at apply time.

Reply:
It's not an issue logically. Just that we have two variables for a single value which is unnecessary. There's a comment in there already about this - https://github.com/stellar/stellar-core/blob/62313d8b1113d219417ca50f9ad09716ad12c69f/src/rust/src/contract.rs#L118.

Anyways, we don't need to block on this.